### PR TITLE
support batch input for boxhead

### PR DESCRIPTION
**Pull request by zhangxiaoli73**
## What changes were proposed in this pull request?
support batch input for boxhead

## How was this patch tested?

unit tests

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/XXX

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/BoxHead.scala [hunk 1]
```diff
@@ -283,26 +284,79 @@ private[nn] class BoxPostProcessor(
     }
     val classLogits = input[Tensor[Float]](1)
     val boxRegression = input[Tensor[Float]](2)
-    val bbox = input[Tensor[Float]](3)
+    require(input(3).isInstanceOf[Table], s"bbox should be table, but gte ${input(3)}")
```
Comment (wzhongyuan) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/BoxHead.scala]: typo

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/BoxHead.scala [hunk 2]
```diff
@@ -146,7 +147,7 @@ private[nn] class BoxPostProcessor(
     val clsScores = selectTensor(scores.select(2, clsInd + 1), inds, 1)
     val clsBoxes = selectTensor(boxes.narrow(2, clsInd * 4 + 1, 4), inds, 1)
 
-    val keepN = nmsTool.nms(clsScores, clsBoxes, nmsThresh, inds)
+    val keepN = nmsTool.nms(clsScores, clsBoxes, nmsThresh, inds, orderWithBbox = true)
```
Comment (wzhongyuan) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/BoxHead.scala]: Bbox -> 'BBox' ?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Nms.scala [hunk 3]
```diff
@@ -61,10 +61,12 @@ class Nms extends Serializable {
    * @param thresh  overlap thresh
    * @param indices buffer to store indices after nms
    * @param sorted whether the scores are sorted
+   * @param orderWithBbox whether return indices in the order with original bbox index
    * @return the length of indices after nms
    */
   def nms(scores: Tensor[Float], boxes: Tensor[Float], thresh: Float,
-    indices: Array[Int], sorted: Boolean = false): Int = {
+    indices: Array[Int], sorted: Boolean = false,
+    orderWithBbox: Boolean = false): Int = {
```
Comment (wzhongyuan) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Nms.scala]: `orderWithBBox` to keep consistent

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/BoxHead.scala [hunk 4]
```diff
@@ -283,26 +284,79 @@ private[nn] class BoxPostProcessor(
...
 
     val classProb = softMax.forward(classLogits)
-    BboxUtil.decodeWithWeight(boxRegression, bbox, weight, boxesBuf)
-
-    val boxesInImage = bbox.size(1)
-    val proposalSplit = boxesBuf.split(boxesInImage, dim = 1)
-    val classProbSplit = classProb.split(boxesInImage, dim = 1)
-
-    val roilabels = filterResults(proposalSplit(0), classProbSplit(0), nClasses)
+    BboxUtil.decodeWithWeight(boxRegression, concatBoxes, weight, boxesBuf)
```
Comment (wzhongyuan) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/BoxHead.scala]: `BBoxUtil` ?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/BoxHead.scala (hunk 1 above)
Comment (zhangxiaoli73) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/BoxHead.scala]: ok, changed

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/BoxHead.scala (hunk 2 above)
Comment (zhangxiaoli73) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/BoxHead.scala]: ok, changed

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Nms.scala (hunk 3 above)
Comment (zhangxiaoli73) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Nms.scala]: ok, changed

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/BoxHead.scala (hunk 4 above)
Comment (zhangxiaoli73) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/BoxHead.scala]: Sorry, it's hard to change as this object has existed for a few release. Users may use in their cases.

**Comment by zhangxiaoli73** [comment]
@wzhongyuan make some changes according to your review comments. Do you have any other suggestions?

**Comment by zhangxiaoli73** [comment]
Jenkins pass: http://mlt-bddl01.sh.intel.com:8080/view/PR/job/BigDL-PRVN-lite/970/

**Comment by wzhongyuan** [comment]
LGTM

**Pull request merged**